pytorch_lightning/utilities/xla_device.py

```diff
@@ -103,3 +105,8 @@ def tpu_device_exists() -> bool:
             if XLADeviceUtils._TPU_AVAILABLE:
                 os.environ["PL_TPU_AVAILABLE"] = '1'
         return XLADeviceUtils._TPU_AVAILABLE
+
+
+def tpu_training_and_local_rank_zero(trainer: 'pl.Trainer') -> bool:
```

Comment:
Why not override `is_global_zero` in tpu_spawn.py with this?

Reply:
It wouldn't be the right thing to do.

```python
def is_global_zero():
	return self.local_rank == 0
```

Reply:
n00b questions: is there no notion of global rank with TPU training?

is this check for global zero only for checkpointing? if so, should we have a property on the trainer that says whether this trainer rank should save checkpoints? for example, with sharded training we'll soon need to save checkpoint shards from each rank because the full model won't fit in single-gpu memory

Reply:
@ananthsub Yes, there is. But for eg., let's consider TPU v3-32, it would require 4 VMs for training, each for 8 cores. As it separate VMs, it is required to save checkpoints on every local rank.

But yeah, I really like the idea of having a property that says on which rank to save checkpoints. Right now, it is hardcoded to global rank zero in `ModelCheckpoint` Callback.

Reply:
@kaushikb11 Worth to add in another PR :)

Reply:
Yup, will create an issue around it.